web-common/src/features/exports/ExportMenu.svelte

```diff
@@ -1,38 +1,54 @@
 <script lang="ts">
   import Button from "@rilldata/web-common/components/button/Button.svelte";
   import * as DropdownMenu from "@rilldata/web-common/components/dropdown-menu";
+  import CaretDownIcon from "@rilldata/web-common/components/icons/CaretDownIcon.svelte";
+  import Export from "@rilldata/web-common/components/icons/Export.svelte";
   import Tooltip from "@rilldata/web-common/components/tooltip/Tooltip.svelte";
   import TooltipContent from "@rilldata/web-common/components/tooltip/TooltipContent.svelte";
-  import Export from "@rilldata/web-common/components/icons/Export.svelte";
   import {
+    createQueryServiceExport,
     V1ExportFormat,
-    type V1MetricsViewAggregationRequest,
+    type V1Query,
   } from "@rilldata/web-common/runtime-client";
-  import { onMount } from "svelte";
-  import type ScheduledReportDialog from "../scheduled-reports/ScheduledReportDialog.svelte";
   import { builderActions, getAttrs } from "bits-ui";
-  import CaretDownIcon from "@rilldata/web-common/components/icons/CaretDownIcon.svelte";
+  import { onMount } from "svelte";
+  import { get } from "svelte/store";
+  import { runtime } from "../../runtime-client/runtime-store";
+  import type TScheduledReportDialog from "../scheduled-reports/ScheduledReportDialog.svelte";
 
   export let disabled: boolean = false;
   export let workspace = false;
   export let label: string;
   export let includeScheduledReport = false;
-  export let queryArgs: V1MetricsViewAggregationRequest | undefined = undefined;
+  export let getQuery: (isScheduled: boolean) => V1Query | undefined; // In the future, `V1Query` should not change depending on whether the query is scheduled or not
```

Comment:
What is the comment in reference to? Isnt the difference in query because of where the export is called from?

Reply:
I was hoping we could replace the `getQuery()` prop with a `query` prop, assuming the `V1Query` would be the same between ad-hoc & scheduled exports. But per your comment above, it seems we need to inject the `end` date for ad-hoc exports. So I'll remove this comment.